src/routes/api/autohooks.ts

```diff
@@ -1,9 +1,13 @@
 import { FastifyInstance } from 'fastify'
 
 export default async function (fastify: FastifyInstance) {
-  fastify.addHook('onRequest', async (request) => {
-    if (!request.url.startsWith('/api/auth/login')) {
-      await request.jwtVerify()
+  fastify.addHook('onRequest', async (request, reply) => {
```

Comment:
I know this is from autoload experience, but it wouldn't be better to put this as a plugin that can be added on-demand; it seems quite counterintuitive the way is set.

Having the possibility of add this to the routes an implementer wants to protect seems to have better ergonomics than a soft check. wdyt?

Reply:
Do you recommend to perform this check to every route that need authentication? A set of controllers?

If yes, everything under `/api` requires to be authenticated anyway, so I think it's much safer to have this check here and protect a large part of the application.

Reply:
I can add a check here: [src/plugins/custom/authorization.ts](https://github.com/fastify/demo/pull/33/files#diff-00db5aed3aea31be2f7e173651cc17e6984158fc638ae2be571df33a65f56d3b)

Reply:
Don't have an opinion on that, I believe that it varies from project to project, org to org, developer to developer; nevertheless having an example of that that showcases how to protect on a per-route basis will be of help for the community as they might face this requirement and require for them to apply this logic.

Reply:
I have already a per-route approach for authorization, but for authenticated users with roles, isn't it enough?